### PR TITLE
fix-android-bugs

### PR DESCRIPTION
**Pull request by husseinelhussein**
fix: use of old APIS, notifications not showing and permissions issues

### CODE REVIEW

**Comment by umuur** [comment]
Could you please assign this to @Samoz93

File: src/android/GeoNotification.java [hunk 1]
```diff
@@ -21,19 +26,31 @@ public class GeoNotification {
...
     }
 
     public Geofence toGeofence() {
-        return new Geofence.Builder()
-            .setRequestId(id)
-            .setTransitionTypes(transitionType)
-            .setCircularRegion(latitude, longitude, radius)
-            .setLoiteringDelay(loiteringDelay == 0 ? 60 * 60 * 1000 : loiteringDelay)
-            .setExpirationDuration(Long.MAX_VALUE).build();
+        if(transitionType == 1 || transitionType == 2){
```
Comment (samoz93) [inline src/android/GeoNotification.java:39]: I am not sure what those numbers represents, can we add comments on the non-obvious part of the code

File: src/android/GeofenceJsEvent.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    public static void onTransitionReceived(List<GeoNotification> notifications) {
+        Log.d(TAG, "Transition Event Received!");
+        String js = "setTimeout('geofence.onTransitionReceived("
+                + Gson.get().toJson(notifications) + ")',0)";
+        sendJavascript(js);
+    }
+
+    public static void onNotificationClicked(String data) {
+        if (data != null) {
+            String js = "setTimeout('geofence.onNotificationClicked(" + data + ")', 100)";
```
Comment (samoz93) [inline src/android/GeofenceJsEvent.java:24]: Nice way to share results with our JS code!
If we're using an easier platform than Cordova I would've suggested sending the payload with the notification and read it again once the user clicked it

File: src/android/GeofenceJsEvent.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+            sendJavascript(js);
+        }
+    }
+
+    public static void onPermissionsResult(boolean result){
+        String js = "setTimeout('geofence.onPermissions(" + result + ")', 300)";
```
Comment (samoz93) [inline src/android/GeofenceJsEvent.java:30]: Same here, We can use callbackContext to send the actions results, rather than manipulating the JS code

**Review by samoz93 — APPROVED** [review]
I will defer to your judgment on this PR, If it tested then we can merge
My note on this PR since our knowledge on native code is fairly limited is to have more comments to better understand all the parts in play 🙌

File: src/android/GeofenceJsEvent.java (hunk 3 above)
Comment (husseinelhussein) [reply src/android/GeofenceJsEvent.java:30]: for some reason `callbackContext` doesn't always work, thats why i added `onPermissions()`

**Pull request merged**